src/main/java/io/jenkins/plugins/analysis/core/steps/IssuesPublisher.java

```diff
@@ -198,7 +200,11 @@ private History createAnalysisHistory(final ResultSelector selector, final Repor
         if (referenceJobName != null) {
             Optional<Job<?, ?>> referenceJob = new JenkinsFacade().getJob(referenceJobName);
             if (referenceJob.isPresent()) {
-                baseline = referenceJob.get().getLastBuild();
+                if (referenceBuildId != null) {
+                    baseline = referenceJob.get().getBuild(referenceBuildId);
+                } else {
+                    baseline = referenceJob.get().getLastBuild();
+                }
             }
```

Comment:
I think I have overseen a bug in my code (that is now in your code as well): the baseline can be `null` (build has been deleted, etc.). So it would make sense to add after the if-else (i.e. for both cases) a 
```
if (baseline == null) {
   return new NullHistory();
}
```

It would also make sense to test that in an additional test.

Reply:
I've added the check, and made two more integration tests:

- use a reference job with no builds
- use a reference job with builds, but referring to a non existent build id

I was sort of expecting these tests to show all issues as "new" since the baseline was missing, but they are "outstanding".  I assume this is intentional.

I *have* had this issue on my local install, and this has caused build failures.  I am slightly worried that with this change, a missing baseline build may not show up as an error at all, and will therefore be hidden.

Reply:
The first build of a job cannot create new warnings, this is intentional, yes.

But is is a good point if this should be ignored or produce an error. Maybe a better way is to use something like 

```
if (referenceBuildId == null) {
   baseline = referenceJob.get().getLastBuild();
   if (baseline == null) {
      throw new IllegalArgumentException(String.format("Reference job '%s' does not contain a valid build", referenceJob);
   }
}
else {
   baseline = referenceJob.get().getBuild(referenceBuildId);
   if (baseline == null) {
      throw new IllegalArgumentException(String.format("Reference job '%s' does not contain build '%s'", 
referenceJob, referenceBuildId );
   }
}
```  
What do you think? Or we can create an error message using `filtered.logError`. That will be flagged as an error triangle at the build page.

Reply:
In this case it's the first build of job that has explicitly been told to compare warnings with another job or a specific build of another job.

If the specific build does not exist, I'd say that's an error.  If you only specified a reference job, and that job does not exist, I'd say that's an error.  If you specify a reference job, and that job does not have any builds to compare warnings to, I'd still say that's an error -- that configuration does not make sense.

I think your suggestion looks good, let me know, and I'll add it to the pull request.

Reply:
What kind of error do you prefer? Using the logger.error method will create something like https://ci.jenkins.io/job/Plugins/job/warnings-ng-plugin/job/reference/1/
(Open Tasks error triangle - that you can click and see the details of the error messages). Or an exception that fails the build? This seems a little bit too hard for me and is not visualized in the build...

Reply:
That's a good job for illustration.

I agree, failing the build would be too hard, the error triangle is the best option.

I hope to amend the pull request tomorrow.

Reply:
Thank you very much!